### PR TITLE
Requirements

### PR DESCRIPTION
**Pull request by MateoCaicedoW**
in this branch the requirements table was created and the foreign keys were added  in their models, I also delete the email duplicate in user model

### CODE REVIEW

File: app/models/departments.go [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package models
+
+import (
+	"time"
+
+	"github.com/gofrs/uuid"
+)
+
+type Departments struct {
```
Comment (BryanMoslo) [inline app/models/departments.go]: Please name this model in Singular.

File: app/models/requirement.go [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+	"github.com/gofrs/uuid"
+)
+
+type Requirement struct {
+	ID                     uuid.UUID `db:"id"`
+	Title                  string    `db:"title"`
+	Description            string    `db:"description"`
+	CreatedBy              uuid.UUID `db:"created_by"`
+	CreatedAt              time.Time `db:"created_at"`
+	RequestingDepartmentId uuid.UUID `db:"requesting_department_id"`
```
Comment (BryanMoslo) [inline app/models/requirement.go]: Acronyms is Go should be in uppercase, it should be RequestingDepartmentID

File: app/models/requirement.go [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+	ServiceDepartmentId    uuid.UUID `db:"service_department_id"`
```
Comment (BryanMoslo) [inline app/models/requirement.go]: Acronyms is Go should be in uppercase.

File: app/models/requirement.go [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+	RequirementTypeId      uuid.UUID `db:"requirement_type_id"`
+	RequirementSubTypeId   uuid.UUID `db:"requirement_sub_type_id"`
```
Comment (BryanMoslo) [inline app/models/requirement.go]: Acronyms is Go should be in uppercase.

**Review by BryanMoslo — COMMENTED** [review]
Please take a look at my suggestions @MateoCaicedoW

**Review by rdiazrovira — CHANGES_REQUESTED** [review]
Multiple columns of the table **"requirements"** should be nullable.
The columns "modified_at", "approved_at", "accepted_at", "declined_at", "proccessed_at", "assigned_at", and "finished_at" must not be set with default values.
Also, please fix the typo "proccessed_at". That should be named "processed_at".

**Review by rdiazrovira — APPROVED** [review]

**Pull request merged**